airflow/models/dag.py

```diff
@@ -2752,8 +2753,18 @@ def add_logger_if_needed(ti: TaskInstance):
         # for task readiness and dependency management. This is notably faster
         # than creating a BackfillJob and allows us to surface logs to the user
         while dr.state == DagRunState.RUNNING:
+            session.expire_all()
```

Comment:
Any reason to do this?

Reply:
I think it's to force sqlalchemy to fetch the new state of all the objects from the DB (force refresh).